pkg/virt-handler/non-root.go

```diff
@@ -117,6 +117,8 @@ func getTapDevices(vmi *v1.VirtualMachineInstance, networkBindings map[string]v1
 	}
 
 	networkNameScheme := namescheme.CreateHashedNetworkNameScheme(vmi.Spec.Networks)
+	networkNameScheme = namescheme.UpdatePrimaryPodIfaceNameFromVMIStatus(networkNameScheme, vmi.Spec.Networks, vmi.Status.Interfaces)
```

Comment:
`namescheme.UpdatePrimaryPodIfaceNameFromVMIStatus` takes a podIfaceNamesByNetworkName map (a naming scheme), and updates its primary pod interface name in case it exists in the VMI.Status.Interfaces slice.